spec/api/sul_bib/authorship_api_spec.rb

```diff
@@ -44,44 +43,24 @@
   # method path, json, headers
   # e.g.
   # let(:http_request) do
-  #   post '/authorship', json_data, headers
+  #   post '/authorship', request_data.to_json, headers
   # end
 
-  shared_examples 'it is successful' do
-    it 'has response code < 300' do
-      http_request # defined in calling context
-      expect(response.status).to be < 300
-    end
-  end
-
-  shared_examples 'it creates a new contribution' do
-    it 'it increases number of contribution records by one' do
-      expect do
-        http_request # defined in calling context
-      end.to change(Contribution, :count).by(1)
-    end
-  end
-
   shared_examples 'it creates new contributions and publications' do
     context 'with no other contributions' do
       let(:request_data) { valid_data_for_post.merge(author_hash) }
-      it_behaves_like 'it is successful'
-      it_behaves_like 'it creates a new contribution'
+      it 'successfully creates one new contribution' do
+        expect { http_request }.to change(Contribution, :count).by(1)
+        expect(response.status).to eq(201)
+      end
     end
```

Comment:
OK, obviously we differ on grouping more than one expect in a spec, but that's not too important here.  The prior success check was obviously generic (any 200+ response is a success).  I suppose it could also use a `response.success?` or something similiar in an rspec dialect.  So long as _everything_ using this shared example must be returning a 201, this change is OK.  I actually don't know if that's reasonable or not for this API endpoint or the specs using this shared example.  It does make the shared example a little less share-able.

Reply:
Yeah, I made sure all the statuses lined up and preferred being exact where possible.  This keeps some of the insane nesting of `shared_examples` down.  

In integration tests in particular, I find one almost always needs a separate `expect` for checking status.  It doesn't make sense to me to have a separate request/response cycle for a complex operation and then just check if `status == 200` and throw it away.